src/bigdecimal.ts

```diff
@@ -1152,12 +1154,12 @@ export class BigDecimal {
                 if (scale !== undefined) {
                     throw new RangeError('You should not give scale when number is a double');
                 }
-
                 return BigDecimal.fromDouble(value, mc);
             }
             if (!(value > Number.MIN_SAFE_INTEGER && value <= Number.MAX_SAFE_INTEGER)) {
-                // Unsafe range, build from bigint
-                return BigDecimal.fromBigInt(BigInt(value), scale, mc);
+                // Unsafe range, build from string
+                value = String(value);
+                return BigDecimal.fromString(value, 0, value.length, scale, mc);
```

Comment:
Something like this: 

```suggestion
                // Unsafe range, build from string
                value = String(value);
                const bigDecimal = BigDecimal.fromString(value, 0, value.length, mc);
                if (scale !== undefined) {
                    return bigDecimal.setScale(scale, mc ? mc.roundingMode : undefined);
                } else {
                    return bigDecimal;
                }
```